model/src/main/kotlin/LicenseFinding.kt

```diff
@@ -57,5 +57,10 @@ data class LicenseFinding @JsonCreator constructor(
     @JsonCreator
     constructor(licenseName: String) : this(licenseName, sortedSetOf())
 
-    override fun compareTo(other: LicenseFinding) = license.compareTo(other.license)
+    override fun compareTo(other: LicenseFinding) =
+            when {
```

Comment:
Why not always compare all attributes ? I think it's easier to see whether it's correct then.
There is a neat library for this which has a builder, same also for hascode and equals which makes it super easy to have those consistent and readable: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/CompareToBuilder.html

Relying in on the hash code is wrong because 
you can have two findings with have same license, different copyrights but same copyright hashcode.
This breaks again the contract between equals and compareTo.

Reply:
If the first compared attribute is different already, why compare the others? The problem is that `SortedSet<String>` has no natural order, so I have added a comparator for that now.